sys/newlib/thread_safe/Makefile.include

```diff
@@ -0,0 +1,22 @@
+# define the compile test file
+define __has_reent_small_test
+#include <sys/reent.h>\n
+#ifndef _REENT_SMALL\n
+#error wasting 1KB\n
+#endif
+endef
+
+# compile the tests
+__has_reent := $(shell echo "\#include <sys/reent.h>\n" | $(CC) -E $(CFLAGS) $(INCLUDES) - 2> /dev/null > /dev/null; echo $$?)
```

Comment:
echo is really the worst way to get a text string in shell scripts. There
is no standard specification and the flags and default behaviour differ
between almost every platform (OSX, Linux, BSD, windows etc.)
Also, the \n is redundant since echo will add a newline to the string (on
all platforms that I know of) when called with no flags.

Den 23 aug. 2016 09:52 skrev "Pieter Willemsen" <notifications@github.com>:

In sys/newlib/thread_safe/Makefile.include
<https://github.com/RIOT-OS/RIOT/pull/4529#discussion_r75816861>:

> @@ -0,0 +1,22 @@
> +# define the compile test file
> +define __has_reent_small_test
> +#include <sys/reent.h>\n
> +#ifndef _REENT_SMALL\n
> +#error wasting 1KB\n
> +#endif
> +endef
> +
> +# compile the tests
> +__has_reent := $(shell echo "\#include <sys/reent.h>\n" | $(CC) -E $(CFLAGS) $(INCLUDES) - 2> /dev/null > /dev/null; echo $$?)

the \n results in an error: "stdin>:1:23: error: extra tokens at end of
#include directive [-Werror]", adding -e to the echo command solves it (
-e: enable interpretation of backslash escapes)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/RIOT-OS/RIOT/pull/4529/files/0b380225290d6b1efeea9a7bbb273684b417672d#r75816861>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AATYQu8sumUmNCJsn-omBi-6etbn_ofjks5qiqa9gaJpZM4G43Px>
.